Functions/xASL_bids_BIDSifyFixBasicFields.m

```diff
@@ -0,0 +1,90 @@
+function jsonIn = xASL_bids_BIDSifyFixBasicFields(jsonIn,studyPar,bidsModality)
+%xASL_bids_BIDSifyFixBasicFields Check if required fields exist in studyPar but not in jsonIn 
+% or if we can find them out in other ways
+%
+% FORMAT: jsonIn = xASL_bids_BIDSifyFixBasicFields(jsonIn,studyPar);
+%
+% INPUT:
+%   jsonIn       - JSON with the input fields - from DICOMs (REQUIRED)
+%   studyPar     - Manually defined parameters (REQUIRED)
+%   bidsModality - Modality (CHAR ARRAY, OPTIONAL, DEFAULT = 'asl')
+%
+% OUTPUT: 
+%   jsonIn    - ordered and checked JSON structure
+%
+% -----------------------------------------------------------------------------------------------------------------------------------------------------
+% DESCRIPTION: Check if required fields exist in studyPar but not in jsonIn or if we can find them in other ways.
+%              
+%              The BIDSification of JSON metadata requires at least some basic fields. If dcm2niix can't extract
+%              fields like Manufacturer from the DICOM data (strict anonymization), we need to be able to read them
+%              from the studyPar JSON (manually inserted). Alternatively we can check other DICOM tags for information.
+%              
+%              This function can potentially be enhanced in future release to fix other fields besides the Manufacturer as well.
+%              To enable this functionality for different modalities, we introduced the bidsModality parameter.
+%              
+%              This function is called by:
+%              
+%              - `xASL_bids_BIDSifyM0`
+%              - `xASL_bids_BIDSifyASLJSON`
+%              - `xASL_bids_BIDSifyAnatJSON`
+%              
+% EXAMPLE:     jsonIn = xASL_bids_BIDSifyFixBasicFields(jsonIn,studyPar,'asl');
+%
+% __________________________________
+% Copyright 2015-2021 ExploreASL
+
+    % Default
+    if nargin < 3
+        bidsModality = 'asl';
+    end
+
+    % List of required fields
+    switch bidsModality
+        case 'asl'
+            fieldList = {'Manufacturer', 'TotalAcquiredPairs', 'FlipAngle'};
+        case 'm0'
+            fieldList = {'Manufacturer', 'FlipAngle'};
+        otherwise
+            fieldList = {'Manufacturer'};
+    end
+    
+    % Iterate over fields
+    for iField = 1:numel(fieldList)
+        if isfield(studyPar,fieldList{iField}) && ~isfield(jsonIn,fieldList{iField})
+            jsonIn.(fieldList{iField}) = studyPar.(fieldList{iField});
+        end
+    end
```

Comment:
@jan-petr: I forgot to say in the dev meeting that this can also help if the TotalAcquiredPairs are in the studyPar already, but not in the metadata. So it's really not only about the Manufacturer. I'm open for suggestions on improvements though. Would be cool if you could check the general behavior though.

Reply:
Yes - I understood that at the meeting already. At first was afraid that this will create a mess. But I have checked all the code and it won't create conflicts - so good to have everything here...

Reply:
It was mainly the function name I didn't like ;) 'Fix' sounds very general - all functions 'fix' something. Besides, this is really more of a 'merge'. what about xASL_bids_MergeStudyPar?

Reply:
Sounds good 4f6c55a 😄